src/server/service/diagnostics/log.rs

```diff
@@ -155,44 +157,57 @@ impl Iterator for LogIterator {
                         continue;
                     }
                 };
-                if input.len() < TIMESTAMP_LENGTH {
+                if self.pre_log.time < self.begin_time && input.len() < TIMESTAMP_LENGTH {
                     continue;
                 }
+                let mut item = LogMessage::default();
                 match parse(&input) {
                     Ok((content, (time, level))) => {
-                        // The remain content timestamp more the end time or this file contains inrecognation formation
-                        if time == INVALID_TIMESTAMP || time > self.end_time {
-                            break;
-                        }
-                        if time < self.begin_time {
-                            continue;
-                        }
-                        if self.level_flag != 0 && self.level_flag & (1 << (level as usize)) == 0 {
-                            continue;
-                        }
-                        if !self.patterns.is_empty() {
-                            let mut not_match = false;
-                            for pattern in self.patterns.iter() {
-                                if !pattern.is_match(content) {
-                                    not_match = true;
-                                    break;
-                                }
-                            }
-                            if not_match {
-                                continue;
-                            }
-                        }
-                        let mut item = LogMessage::default();
+                        self.pre_log.set_time(time);
+                        self.pre_log.set_level(level);
+
                         item.set_time(time);
                         item.set_level(level);
                         item.set_message(content.to_owned());
-                        return Some(item);
                     }
                     Err(err) => {
                         warn!("parse line failed: {:?}", err);
```

Comment:
Maybe we no longer needs to output warnings (considering that we are now accepting such logs)?

Reply:
I think so, will remove it soon.